### PR TITLE
Aarch (follow-up on #166)

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
I was able to successfully build qt on conda-forge's arm server. Seeing now whether it builds also fine on the CI servers.

A few caveats:
- I needed to create https://github.com/conda-forge/qt-feedstock/blob/55695bb5686bed85c57d157589f4889d2c4f583c/recipe/patches/disable_skia_use_expat.patch to disable expat deep inside qtwebengine, as for some reason there it couldn't find expat.h. I have no idea whether or not that makes any practical difference.
- I needed to add `QT += webengine` to the hello.pro test file, as otherwise there were undefined references to pcre stuff. Again I don't know whether this will make a practical difference in e.g. cmake builds.
- The PR doesn't add ppc64 - not sure whether there is interest for it, I guess this could be done later.

**Comment by Tobias-Fischer** [comment]
Yeah I'm not sure what's going on with these libffi errors - does anyone have a clue? he1b5a44_1007 is from the old libffi3.2.1, whereas h58526e2_2 is from 3.3. Recently this happened, not sure if related: https://github.com/conda-forge/conda-forge-pinning-feedstock/commit/c6f978211ad8abe79ec841d05e967d9d9665eab7#diff-ff61408cdc05bc9667deeadb55e4aaceb1371972076b6bf6934f9008920f2bd2

```
libffi: ValueError: Incompatible component merge: 
- 'he1b5a44_1007'
- 'h58526e2_2'
```

/cc @wolfv @isuruf @hmaarrfk

**Comment by Tobias-Fischer** [comment]
Hi @conda-forge/qt, I think this is ready for review. I'm not sure how you typically built this feedstock as it times out on the CIs. It built successfully on conda-forge's ARM server. Please see my caveats above, but I think none of them should stop us from having QT on aarch :)

**Comment by hmaarrfk** [comment]
i'm confused as to why 
```
The reported errors are:
⇟   Encountered problems while solving.
   Problem: cannot install both ld_impl_linux-aarch64-2.35.1-h36823c2_1 and ld_impl_linux-aarch64-2.35.1-h02ad14f_1
   Problem: cannot install both binutils_impl_linux-aarch64-2.35.1-h339fa7d_1 and binutils_impl_linux-aarch64-2.35.1-h026399b_1
```

this is happening. i'm going to try to rebuild the libffi on the cf server.

**Comment by hmaarrfk** [comment]
if we remove the host and build sections, and just combine them manually, we get.

```
## Package Plan ##

  environment location: /home/hmaarrfk/mambaforge/conda-bld/qt_1612127032399/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho
ld_placehold_placehold_placehold_placehold_placehold_placehold_place


The following NEW packages will be INSTALLED:

    _openmp_mutex:                     4.5-1_gnu                 conda-forge
    alsa-lib-cos7-aarch64:             1.1.8-h9d98e8f_1104       conda-forge
    alsa-lib-devel-cos7-aarch64:       1.1.8-h9d98e8f_1104       conda-forge
    atk-cos7-aarch64:                  2.28.1-h9d98e8f_1104      conda-forge
    atk-devel-cos7-aarch64:            2.28.1-h9d98e8f_1104      conda-forge
    binutils_impl_linux-aarch64:       2.35.1-h026399b_1         conda-forge
    binutils_linux-aarch64:            2.35-h0a73b0c_29          conda-forge
    bison:                             3.4-he1b5a44_1            conda-forge
    ca-certificates:                   2020.12.5-h4fd8a4c_0      conda-forge
    cairo-cos7-aarch64:                1.15.12-h9d98e8f_1104     conda-forge
    cairo-devel-cos7-aarch64:          1.15.12-h9d98e8f_1104     conda-forge
    certifi:                           2020.12.5-py39ha65689a_1  conda-forge
    clang:                             11.0.1-h8af1aa0_1         conda-forge
    clang-11:                          11.0.1-default_h4839fef_1 conda-forge
    clang-tools:                       11.0.1-default_h4839fef_1 conda-forge
    clangdev:                          11.0.1-default_h4839fef_1 conda-forge
    clangxx:                           11.0.1-default_h6215f28_1 conda-forge
    cups-devel-cos7-aarch64:           1.6.3-h9d98e8f_1104       conda-forge
    cups-libs-cos7-aarch64:            1.6.3-h9d98e8f_1104       conda-forge
    dbus:                              1.13.6-h32d69cc_1         conda-forge
    expat:                             2.2.10-h3621d8e_0         conda-forge
    flex:                              2.6.4-h38784ca_1004       conda-forge
    fontconfig:                        2.13.1-hd44bfdd_1004      conda-forge
    fontconfig-cos7-aarch64:           2.13.0-h9d98e8f_1104      conda-forge
    fontconfig-devel-cos7-aarch64:     2.13.0-h9d98e8f_1104      conda-forge
    freetype:                          2.10.4-hdf53a3c_1         conda-forge
    freetype-cos7-aarch64:             2.8-h9d98e8f_1104         conda-forge
    freetype-devel-cos7-aarch64:       2.8-h9d98e8f_1104         conda-forge
    freetype-devel-cos7-aarch64:       2.8-h9d98e8f_1104         conda-forge
    fribidi-cos7-aarch64:              1.0.2-h9d98e8f_1104       conda-forge
    gcc_impl_linux-aarch64:            9.3.0-h5b547fa_17         conda-forge
    gcc_linux-aarch64:                 9.3.0-h3f8fa26_29         conda-forge
    gettext:                           0.19.8.1-h4454c44_1005    conda-forge
    glib:                              2.66.4-h0119ad3_2         conda-forge
    glib-tools:                        2.66.4-h0119ad3_2         conda-forge
    glib2-cos7-aarch64:                2.56.1-h9d98e8f_1104      conda-forge
    glib2-devel-cos7-aarch64:          2.56.1-h9d98e8f_1104      conda-forge
    gmp:                               6.2.1-h7fd3ca4_0          conda-forge
    gperf:                             3.1-h7fd3ca4_0            conda-forge
    gst-plugins-base:                  1.18.3-hf9c081e_0         conda-forge
    gstreamer:                         1.18.3-h68f305b_0         conda-forge
    gtk2-cos7-aarch64:                 2.24.31-h9d98e8f_1104     conda-forge
    gtk2-devel-cos7-aarch64:           2.24.31-h9d98e8f_1104     conda-forge
    gtkmm24-cos7-aarch64:              2.24.5-h9d98e8f_1104      conda-forge
    gtkmm24-devel-cos7-aarch64:        2.24.5-h9d98e8f_1104      conda-forge
    gxx_impl_linux-aarch64:            9.3.0-he181498_17         conda-forge
    gxx_linux-aarch64:                 9.3.0-h235e3a5_29         conda-forge
    harfbuzz-cos7-aarch64:             1.7.5-h9d98e8f_1104       conda-forge
    icu:                               68.1-h884eca8_0           conda-forge
    jpeg:                              9d-hfd2af3c_0             conda-forge
    kernel-headers-cos7-aarch64:       4.18.0-h9d98e8f_1104      conda-forge
    kernel-headers_linux-aarch64:      4.18.0-hca5cc2b_10        conda-forge
    krb5:                              1.17.2-he08ec14_0         conda-forge
    ld_impl_linux-aarch64:             2.35.1-h02ad14f_1         conda-forge
    libclang:                          11.0.1-default_h4839fef_1 conda-forge
    libclang-cpp:                      11.0.1-default_h4839fef_1 conda-forge
    libclang-cpp11:                    11.0.1-default_h4839fef_1 conda-forge
    libdrm-cos7-aarch64:               2.4.97-h9d98e8f_1104      conda-forge
    libdrm-devel-cos7-aarch64:         2.4.97-h9d98e8f_1104      conda-forge
    libedit:                           3.1.20191231-he28a2e2_2   conda-forge
    libevent:                          2.1.10-hcdb4288_3         conda-forge
    libffi:                            3.3-h884eca8_2            conda-forge
    libgcc-devel_linux-aarch64:        9.3.0-h2e979a9_17         conda-forge
    libgcc-ng:                         9.3.0-h8ab15db_17         conda-forge
    libglib:                           2.66.4-he271a7a_2         conda-forge
    libglvnd-cos7-aarch64:             1.0.1-h9d98e8f_1104       conda-forge
    libglvnd-glx-cos7-aarch64:         1.0.1-h9d98e8f_1104       conda-forge
    libgomp:                           9.3.0-h8ab15db_17         conda-forge
    libiconv:                          1.16-h6dd45c4_0           conda-forge
    libllvm11:                         11.0.1-h6293a0b_0         conda-forge
    libpng:                            1.6.37-hed695b0_2         conda-forge
    libpq:                             12.3-h472cb92_3           conda-forge
    libselinux-cos7-aarch64:           2.5-h9d98e8f_1104         conda-forge
    libselinux-devel-cos7-aarch64:     2.5-h9d98e8f_1104         conda-forge
    libsepol-cos7-aarch64:             2.5-h9d98e8f_1104         conda-forge
    libsepol-devel-cos7-aarch64:       2.5-h9d98e8f_1104         conda-forge
    libstdcxx-devel_linux-aarch64:     9.3.0-h96f9e0b_17         conda-forge
    libstdcxx-ng:                      9.3.0-h7a64470_17         conda-forge
    libthai-cos7-aarch64:              0.1.14-h9d98e8f_1104      conda-forge
    libtiff-cos7-aarch64:              4.0.3-h9d98e8f_1104       conda-forge
    libuuid:                           2.32.1-hf897c2e_1000      conda-forge
    libx11-common-cos7-aarch64:        1.6.7-h9d98e8f_1104       conda-forge
    libx11-cos7-aarch64:               1.6.7-h9d98e8f_1104       conda-forge
    libx11-devel-cos7-aarch64:         1.6.7-h9d98e8f_1104       conda-forge
    libxau-cos7-aarch64:               1.0.8-h9d98e8f_1104       conda-forge
    libxau-devel-cos7-aarch64:         1.0.8-h9d98e8f_1104       conda-forge
    libxcb:                            1.13-h3557bc0_1003        conda-forge
    libxcb-cos7-aarch64:               1.13-h9d98e8f_1104        conda-forge
    libxcomposite-cos7-aarch64:        0.4.4-h9d98e8f_1104       conda-forge
    libxcomposite-devel-cos7-aarch64:  0.4.4-h9d98e8f_1104       conda-forge
    libxcursor-cos7-aarch64:           1.1.15-h9d98e8f_1104      conda-forge
    libxcursor-devel-cos7-aarch64:     1.1.15-h9d98e8f_1104      conda-forge
    libxdamage-cos7-aarch64:           1.1.4-h9d98e8f_1104       conda-forge
    libxdamage-devel-cos7-aarch64:     1.1.4-h9d98e8f_1104       conda-forge
    libxext-cos7-aarch64:              1.3.3-h9d98e8f_1104       conda-forge
    libxext-devel-cos7-aarch64:        1.3.3-h9d98e8f_1104       conda-forge
    libxfixes-cos7-aarch64:            5.0.3-h9d98e8f_1104       conda-forge
    libxfixes-devel-cos7-aarch64:      5.0.3-h9d98e8f_1104       conda-forge
    libxft-cos7-aarch64:               2.3.2-h9d98e8f_1104       conda-forge
    libxi-cos7-aarch64:                1.7.9-h9d98e8f_1104       conda-forge
    libxi-devel-cos7-aarch64:          1.7.9-h9d98e8f_1104       conda-forge
    libxkbcommon:                      1.0.3-he09c752_0          conda-forge
    libxml2:                           2.9.10-hd674cf7_3         conda-forge
    libxrandr-cos7-aarch64:            1.5.1-h9d98e8f_1104       conda-forge
    libxrandr-devel-cos7-aarch64:      1.5.1-h9d98e8f_1104       conda-forge
    libxrender-cos7-aarch64:           0.9.10-h9d98e8f_1104      conda-forge
    libxrender-devel-cos7-aarch64:     0.9.10-h9d98e8f_1104      conda-forge
    libxscrnsaver-cos7-aarch64:        1.2.2-h9d98e8f_1104       conda-forge
    libxscrnsaver-devel-cos7-aarch64:  1.2.2-h9d98e8f_1104       conda-forge
    libxtst-cos7-aarch64:              1.2.3-h9d98e8f_1104       conda-forge
    libxtst-devel-cos7-aarch64:        1.2.3-h9d98e8f_1104       conda-forge
    libxxf86vm-cos7-aarch64:           1.1.4-h9d98e8f_1104       conda-forge
    libxxf86vm-devel-cos7-aarch64:     1.1.4-h9d98e8f_1104       conda-forge
    llvm-tools:                        11.0.1-h6293a0b_0         conda-forge
    llvmdev:                           11.0.1-h6293a0b_0         conda-forge
    lz4-c:                             1.9.3-h01db608_0          conda-forge
    m4:                                1.4.18-h516909a_1001      conda-forge
    make:                              4.3-h309ac5b_1            conda-forge
    mesa-dri-drivers-cos7-aarch64:     18.3.4-h9d98e8f_1104      conda-forge
    mesa-khr-devel-cos7-aarch64:       18.3.4-h9d98e8f_1104      conda-forge
    mesa-libegl-cos7-aarch64:          18.3.4-h9d98e8f_1104      conda-forge
    mesa-libegl-devel-cos7-aarch64:    18.3.4-h9d98e8f_1104      conda-forge
    mesa-libgbm-cos7-aarch64:          18.3.4-h9d98e8f_1104      conda-forge
    mesa-libgl-cos7-aarch64:           18.3.4-h9d98e8f_1104      conda-forge
    mesa-libgl-devel-cos7-aarch64:     18.3.4-h9d98e8f_1104      conda-forge
    mesa-libglapi-cos7-aarch64:        18.3.4-h9d98e8f_1104      conda-forge
    mysql-common:                      8.0.23-h8af1aa0_0         conda-forge
    mysql-devel:                       8.0.23-h8af1aa0_0         conda-forge
    mysql-libs:                        8.0.23-hbb91af6_0         conda-forge
    ncurses:                           6.2-h7fd3ca4_4            conda-forge
    ninja:                             1.10.2-hd62202e_0         conda-forge
    nspr:                              4.29-he1b5a44_1           conda-forge
    nss:                               3.61-hdd3d7a4_0           conda-forge
    openssl:                           1.1.1i-hf897c2e_0         conda-forge
    pango-cos7-aarch64:                1.42.4-h9d98e8f_1104      conda-forge
    pango-devel-cos7-aarch64:          1.42.4-h9d98e8f_1104      conda-forge
    pciutils-cos7-aarch64:             3.5.1-h9d98e8f_1104       conda-forge
    pciutils-devel-cos7-aarch64:       3.5.1-h9d98e8f_1104       conda-forge
    pciutils-libs-cos7-aarch64:        3.5.1-h9d98e8f_1104       conda-forge
    pcre:                              8.44-he1b5a44_0           conda-forge
    perl:                              5.32.0-hb9de7d4_0         conda-forge
    pip:                               21.0.1-pyhd8ed1ab_0       conda-forge
    pkg-config:                        0.29.2-hb9de7d4_1008      conda-forge
    postgresql:                        12.3-hfb687e6_3           conda-forge
    pthread-stubs:                     0.4-hb9de7d4_1001         conda-forge
    python:                            3.9.1-h3bd6a85_5_cpython  conda-forge
    python_abi:                        3.9-1_cp39                conda-forge
    readline:                          8.0-h607064a_2            conda-forge
    ruby:                              2.7.2-h9a605c1_3          conda-forge
    setuptools:                        49.6.0-py39ha65689a_3     conda-forge
    sqlite:                            3.34.0-h43e6a2a_0         conda-forge
    sysroot_linux-aarch64:             2.17-h4744778_10          conda-forge
    tk:                                8.6.10-he2b207e_1         conda-forge
    tzcode:                            2021a-hf897c2e_0          conda-forge
    tzdata:                            2021a-he74cb21_0          conda-forge
    wheel:                             0.36.2-pyhd3deb0d_0       conda-forge
    xorg-libxau:                       1.0.9-hda93590_0          conda-forge
    xorg-libxdmcp:                     1.1.3-h3557bc0_0          conda-forge
    xorg-x11-proto-devel-cos7-aarch64: 2018.4-h9d98e8f_1104      conda-forge
    xz:                                5.2.5-h6dd45c4_1          conda-forge
    yaml:                              0.2.5-h516909a_0          conda-forge
    zlib:                              1.2.11-h516909a_1009      conda-forge
    zstd:                              1.4.8-h41fb7a4_1          conda-forge
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -132,6 +143,7 @@ requirements:
     # Otherwise build and host might end up with two different versions
     # If they do, merge_build_host will fail.
     - readline                           # [linux]
+    - libffi 3.2.1
```
Comment (hmaarrfk) [inline recipe/meta.yaml:146]:
```suggestion
    # hmaarrfk /  Tobias-Fischer 2021/01/31 : 
    # recent rebuilt of binutils didn't bump build number causing conflicts
    # https://github.com/conda-forge/binutils-feedstock/issues/48
    - binutils_impl_linux-aarch64 2.35.1 h026399b_1
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -176,11 +188,12 @@ requirements:
     - llvmdev
     - clangdev
     - libclang
+    - libffi 3.2.1
```
Comment (hmaarrfk) [inline recipe/meta.yaml:191]:
```suggestion
    # hmaarrfk /  Tobias-Fischer 2021/01/31 : 
    # recent rebuilt of binutils didn't bump build number causing conflicts
    # https://github.com/conda-forge/binutils-feedstock/issues/48
    - binutils_impl_linux-aarch64 2.35.1 h026399b_1
```

**Comment by Tobias-Fischer** [comment]
Seems like it's still failing ..

**Comment by Tobias-Fischer** [comment]
By the way, when I tried to compile pyqt, I get the same error as when trying to compile the `hello.pro` test without adding `QT += webengine`. So something still needs fixing unfortunately - any clues?

Here is the error:
```
/home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_build_env/bin/../lib/gcc/aarch64-conda-linux-gnu/9.3.0/../../../../aarch64-conda-linux-gnu/bin/ld: warning: libpcre.so.1, needed by /home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_build_env/bin/../aarch64-conda-linux-gnu/sysroot/usr/lib64/libgthread-2.0.so.0, not found (try using -rpath or -rpath-link)
/home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_build_env/bin/../lib/gcc/aarch64-conda-linux-gnu/9.3.0/../../../../aarch64-conda-linux-gnu/bin/ld: /home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_h_env/qtwebengine/src/3rdparty/chromium/third_party/breakpad/breakpad/src/lib/./libglib-2.0.so.0: undefined reference to `pcre_exec'
/home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_build_env/bin/../lib/gcc/aarch64-conda-linux-gnu/9.3.0/../../../../aarch64-conda-linux-gnu/bin/ld: /home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_h_env/qtwebengine/src/3rdparty/chromium/third_party/breakpad/breakpad/src/lib/./libglib-2.0.so.0: undefined reference to `pcre_study'
/home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_build_env/bin/../lib/gcc/aarch64-conda-linux-gnu/9.3.0/../../../../aarch64-conda-linux-gnu/bin/ld: /home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_h_env/qtwebengine/src/3rdparty/chromium/third_party/breakpad/breakpad/src/lib/./libglib-2.0.so.0: undefined reference to `pcre_dfa_exec'
/home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_build_env/bin/../lib/gcc/aarch64-conda-linux-gnu/9.3.0/../../../../aarch64-conda-linux-gnu/bin/ld: /home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_h_env/qtwebengine/src/3rdparty/chromium/third_party/breakpad/breakpad/src/lib/./libglib-2.0.so.0: undefined reference to `pcre_fullinfo'
/home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_build_env/bin/../lib/gcc/aarch64-conda-linux-gnu/9.3.0/../../../../aarch64-conda-linux-gnu/bin/ld: /home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_h_env/qtwebengine/src/3rdparty/chromium/third_party/breakpad/breakpad/src/lib/./libglib-2.0.so.0: undefined reference to `pcre_get_stringnumber'
/home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_build_env/bin/../lib/gcc/aarch64-conda-linux-gnu/9.3.0/../../../../aarch64-conda-linux-gnu/bin/ld: /home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_h_env/qtwebengine/src/3rdparty/chromium/third_party/breakpad/breakpad/src/lib/./libglib-2.0.so.0: undefined reference to `pcre_compile2'
/home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_build_env/bin/../lib/gcc/aarch64-conda-linux-gnu/9.3.0/../../../../aarch64-conda-linux-gnu/bin/ld: /home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_h_env/qtwebengine/src/3rdparty/chromium/third_party/breakpad/breakpad/src/lib/./libglib-2.0.so.0: undefined reference to `pcre_get_stringtable_entries'
/home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_build_env/bin/../lib/gcc/aarch64-conda-linux-gnu/9.3.0/../../../../aarch64-conda-linux-gnu/bin/ld: /home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_h_env/qtwebengine/src/3rdparty/chromium/third_party/breakpad/breakpad/src/lib/./libglib-2.0.so.0: undefined reference to `pcre_free'
/home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_build_env/bin/../lib/gcc/aarch64-conda-linux-gnu/9.3.0/../../../../aarch64-conda-linux-gnu/bin/ld: /home/Tobias-Fischer/miniforge3/conda-bld/pyqt_split_1612130882746/_h_env/qtwebengine/src/3rdparty/chromium/third_party/breakpad/breakpad/src/lib/./libglib-2.0.so.0: undefined reference to `pcre_config'
```

**Comment by isuruf** [comment]
you probably need a pcre CDT for `libgthread`. Please open an issue in https://github.com/conda-forge/cdt-builds/issues

**Comment by Tobias-Fischer** [comment]
Thanks @isuruf - done. Just for me to understand, why does adding `QT += webengine` fix the problem?

**Comment by Tobias-Fischer** [comment]
Update: I am now trying to build qt with the system (i.e. conda-forge) pcre2; let's see whether this solves the problem. Related PR: https://github.com/conda-forge/pcre2-feedstock/pull/26

**Comment by hmaarrfk** [comment]
Did the binutils conflict get resolved?

**Comment by Tobias-Fischer** [comment]
Nope .. still failing :(. However, it is not failing on the conda-forge ARM machine. Any clue what the difference would be?

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
Seems like it did the trick after all @hmaarrfk :)

**Comment by hmaarrfk** [comment]
I guess we are still waiting on the cdt.

I

**Comment by Tobias-Fischer** [comment]
Yeah .. I still don't understand why things work if I pull in qtwebengine (without the CDT). Any ideas?

**Comment by hmaarrfk** [comment]
its probably linking ot one of its own libraries, instead of using a system library call???

**Comment by hmaarrfk** [comment]
I'm not sure why we need the CDT. it is in the libgthread package

image

File: recipe/meta.yaml [hunk 3]
```diff
@@ -176,12 +188,14 @@ requirements:
     - llvmdev
     - clangdev
     - libclang
+    - pcre2
```
Comment (hmaarrfk) [inline recipe/meta.yaml:191]: why did you add pcre2?

Comment (hmaarrfk) [reply recipe/meta.yaml:191]: i'm trying to build qt without pcre2 on the server. i'll let you know the progress tomorrow i guess.

Comment (Tobias-Fischer) [reply recipe/meta.yaml:191]: Hi @hmaarrfk,
I first tried building it without pcre2. This failed with the linker error in https://github.com/conda-forge/qt-feedstock/pull/181#issuecomment-770462172

I thought I might resolve this linker issue by using the `system-pcre` flag instead of the `qt-pcre` flag. This requires pcre2 with this patch: https://github.com/conda-forge/pcre2-feedstock/pull/26

However, the linker error is still the same. The linker error relates to pcre(1) rather than pcre2, so using the `system-pcre` flag instead of the `qt-pcre` flag doesn't make any difference. It must be something else.

Re CDT/libgthread: I don't understand enough about this to have an informed opinion. My best guess is that somewhere within the QT webengine it uses the correct linking command to pull in pcre, while it doesn't do it by just using QT core.

I uploaded the built qt package to https://anaconda.org/robostack/qt

Installing that in an environment and then running the tests might help you understand what's going on.

**Comment by isuruf** [comment]
As I said earlier you need the pcre2 CDT which is a dependency of libgthread CDT (not the conda package) which is a dependnecy of X11 packages. qtwebengine pulls in chromium which has its own version of pcre2 and therefore the symbols in there satisfy it. So, we should build the pcre2 CDT instead of chasing wild goose.

**Comment by hmaarrfk** [comment]
looks like we are in sync @isuruf https://github.com/conda-forge/cdt-builds/pull/43

THis CDT only exists for cos7, so we would need to add a selector.

File: recipe/meta.yaml (hunk 3 above)
Comment (hmaarrfk) [reply recipe/meta.yaml:191]: thanks. i killed the build.

**Comment by hmaarrfk** [comment]
Is anybody building on the server?

**Comment by Tobias-Fischer** [comment]
@hmaarrfk - not me. Do you want to give it a shot?

**Comment by hmaarrfk** [comment]
started just now.

**Comment by hmaarrfk** [comment]
this still going strong.

Is this ready to merge otherwise? did you need to renable win+osx?

**Comment by Tobias-Fischer** [comment]
Nice! I'll do a cleanup of the PR tomorrow. Only question remaining is whether it's desirable to use system-pcre instead of qt-pcre? I guess it'll reduce the compile time and binary size a little, and compiles just fine, so there is no disadvantage of the system version?

**Comment by hmaarrfk** [comment]
I think systme is ok

**Comment by Tobias-Fischer** [comment]
Great. Did the build finish? It stalled a few times for me.

**Comment by isuruf** [comment]
>  Only question remaining is whether it's desirable to use system-pcre instead of qt-pcre?

Definitely `qt-pcre` unless somebody steps up to do qt builds for all OSes every time there is a pcre version update.

**Comment by hmaarrfk** [comment]
maybe it stalled for me too.

**Comment by Tobias-Fischer** [comment]
Okay - let me try again. I cleaned up the repo and will push here if the build succeeds.

**Comment by Tobias-Fischer** [comment]
Good news: pcre is working all fine now.

Bad news: test fails with `./hello: error while loading shared libraries: libGL.so.1: cannot open shared object file: No such file or directory` - might be an issue with the aarch cdt but I'm not sure. The `mesa-libgl-devel-cos6-x86_64` contains the libGL.so; whereas the `mesa-libgl-devel-cos7-aarch64` does not contain a libGL.so at all, and only contains a symlink without destination of a libglapi.so. However, the `libglvnd-glx-cos7-aarch64` contains the required libGL.so, but even after installing that package, it still does not find the shared library.

Any hints?

**Comment by hmaarrfk** [comment]
@Tobias-Fischer are you trying to install this on the server? I guess we need to ask @isuruf I tried to use docker on the server, it didn't work for me.

**Comment by Tobias-Fischer** [comment]
Yes, I'm talking about the conda-forge server. It would be nice to see if the tests pass I guess ..

**Comment by isuruf** [comment]
mesa-libgl should be available now on the server

**Comment by Tobias-Fischer** [comment]
Many thanks @isuruf! The "hello" test works fine now. The qtwebengine test requires the following libraries, could you please install them as well?
`yum install alsa-lib libXScrnSaver libXtst libXrender-devel libXcursor libXcomposite libXi`

**Comment by Tobias-Fischer** [comment]
Great - all good now, thanks @isuruf. Not sure what the best way is to proceed from here - I have the .tar.bz2 file, should I send it to someone who has write access to the qt repo?

File: recipe/build.sh [hunk 4]
```diff
@@ -28,6 +28,7 @@ USED_BUILD_PREFIX=${BUILD_PREFIX:-${PREFIX}}
 echo USED_BUILD_PREFIX=${BUILD_PREFIX}
 
 MAKE_JOBS=$CPU_COUNT
+export NINJAFLAGS="-j${MAKE_JOBS}"
```
Comment (hmaarrfk) [inline recipe/build.sh:31]: is this line necessary? Is this the new best practice?

**Review by hmaarrfk — COMMENTED** [review]
tiny question above

File: conda-forge.yml [hunk 5]
```diff
@@ -3,3 +3,4 @@ azure:
   settings_win:
     variables:
       CONDA_BLD_PATH: "C:\\bld\\"
+provider: {linux_aarch64: default, linux_ppc64le: default}
```
Comment (hmaarrfk) [inline conda-forge.yml:6]:
```suggestion
# qt 5.12 claims to not support ppc64le. It likely needs patching
# to get it through the build process
provider: {linux_aarch64: default}
```

File: recipe/build.sh (hunk 4 above)
Comment (Tobias-Fischer) [reply recipe/build.sh:31]: qtwebengine uses ninja as build tool. @isuruf and I observed that building qtwebengine still uses all the cores by default. This line should avoid this from happening.

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
Hi @hmaarrfk - quick ping. Is there anything else I can do to get this PR merged? :)

**Comment by isuruf** [comment]
Thanks for working on this. I'll merge this if you commit to building this package on the server manually every time there's a change. (I can copy the package over)

**Comment by wolfv** [comment]
@isuruf how about installing an Azure Worker on the server, and changing smithy so that it uses the appropriate worker pool? We've been using that with a Graviton instance on the RoboStack quite successfully, and it's not a big change in the conda-forge infrastructure.

That way no-one would need to build/copy this manually.

One thing to note is that we need to figure out how to allow only e.g. the Qt repo to use that Agent pool.

**Comment by Tobias-Fischer** [comment]
Hi @isuruf,
I'd be happy to build the package if things change, it's easy enough now that I know what to do. @wolfv's suggestion is interesting though in the long-run I guess, but either way I'd be happy.

**Comment by Tobias-Fischer** [comment]
Another ping @isuruf - as said above, would be happy to rebuild if needed in the future.

**Comment by isuruf** [comment]
Please add yourself as a maintainer

**Comment by Tobias-Fischer** [comment]
Done @isuruf

**Comment by isuruf** [comment]
@conda-forge-admin, rerender
